opam

```diff
@@ -0,0 +1,10 @@
+opam-version: "1.2"
+name: "Dedukti"
+version: "2.5.1"
+maintainer: "dedukti-dev@inria.fr"
+author: "Ronan Saillard"
+homepage: "http://dedukti.gforge.inria.fr"
+bug-reports: "https://gforge.inria.fr/tracker/?func=browse&group_id=4434&atid=13818"
+dev-repo: "https://gforge.inria.fr/git/dedukti/dedukti.git#develop"
+available: [ ocaml-version >= "4.02" ]
+depends: ["menhir" "ocamlbuild"]
```

Comment:
The opam file lacks the `build`and `install` fields

Reply:
Those fields seemed useless for Travis.
The build succeed in : https://travis-ci.org/Deducteam/Dedukti/builds/310058285 without it

Reply:
But if those fields are important for some other applications (and if you know what must be specified in it) feel free to add them

Reply:
Sure, but if one ever wants to pin dedukti with opam (in order to use a dev version, but still keep it managed by opam for things such as updates, etc...), then the fields are needed. So as long as the opam file is added, let's make sure that it is complete and actually usable in practice.

Reply:
The fields are actually very simple (see https://github.com/Gbury/mSAT/blob/master/opam#L7 for instance), but they require to know the correct makefile targets needed for build and installation (which I'm not sure about since I don't actually use the dedukti repo). Additionally, the opam config setting should be respected for things such as binaries. It doesn't seem to be easy with the current makefile as the destination directory is harcoded in the makefile. As a reference, in one of my projects, I have:
```
build: [
  [make "bin"]
]
build-test: [
  [make "test"]
]
install:[
  [make
    "MANDIR=%{man}%"
    "BINDIR=%{archsat:bin}%"
    "SHAREDIR=%{archsat:share}%"
    "install"]
]
remove: [
  [make
    "MANDIR=%{man}%"
    "BINDIR=%{archsat:bin}%"
    "SHAREDIR=%{archsat:share}%"
    "uninstall"]
]
```

Reply:
I must confess I don't understand what "pin dedukti with opam mean".
Further more, is it sufficient to copy and paste the Makefile, unrolling the for loops and adding quotation marks around every words ?
If it is, I can easily do it, but I preferred not doing anything rather that adding non-working code, furthermore for parts of the code Travis is not checking.

Reply:
Rather than copy-pasting the makefile (which is ugly, error-prone, and will definitely break in the future), the opam file should simply call the relvant make targets, passing the correct options (either through a configure script, or environment variables as in my example above).
For information about pining with opa: https://opam.ocaml.org/doc/Usage.html#opam-pin

Reply:
Does something as simple as : 
```
build: [make "all"]
build-test: [make "tests"]
install: [make "install"]
remove: [make "uninstall"]
```
could work ?

Reply:
Almost. The problem is that using just `make install`, the binaries would be installed in the `/usr/bin` directory as harcoded in the makefile, whereas they should be copied in the directory available through the `%{package_name:share}%` variable in opam file (which is usually something like `$(HOME)/.opam/switch/bin/package_name/`).

Reply:
if I put in Makefile : 
```
ifeq ($(INSTALL_DIR),)
INSTALL_DIR=/usr/bin
```
and in opam file : 
```
install:[
  [make
    "INSTALL_DIR=%{package_name:share}%"
    "install"]
]
```
Is it ok for you ?

Reply:
Seems right. It should also be the same for uninstall.

Reply:
I made the two changes suggested in the las two messages. Travis accepted those changes, but I don't think it proves anything, since it was accepted when the opam file did not contain those fields.
https://travis-ci.org/Deducteam/Dedukti/builds/311233385